### ISSUE TITLE
Initial value of country

### ISSUE THREAD

**Issue by TungWang** [natintosh/intl_phone_number_input#162]
**Describe the bug**
When i set initial value, the country never changed.

**Package version**
^0.6.0

**Flutter version**
1.22.4

**To Reproduce**
1. Setup phone number
`_phoneNumber = PhoneNumber(
          phoneNumber: '928825720',
          isoCode: '+886',
          dialCode: '+886',
        );`
2. Setup initial value
`InternationalPhoneNumberInput(
                    countries: ['TW', 'US', 'JP'],
                    initialValue: _phoneNumber,
                    textFieldController: _phoneTextEditingController,
                    selectorConfig: SelectorConfig(
                      selectorType: PhoneInputSelectorType.DROPDOWN,
                      countryComparator: (c1, c2) {
                        return c1.name.compareTo(c2.name);
                      }
                    ),
                    onInputChanged: (PhoneNumber value) {
                      _phoneNumber = value;
                    },
                  ),`

**Screenshots**
<img width="305" alt="截圖 2021-01-12 下午4 22 48" src="https://user-images.githubusercontent.com/45662066/104288193-6fd54a00-54f2-11eb-8bfd-9b6af2eafa88.png">

** Targeted platforms (please complete the following information):**
 - Android, iOS

**Comment by bbarwik** [comment]
It should be
```
                initialValue: PhoneNumber(isoCode: "DE")
```

**Comment by RicardoRB** [comment]
This PR should fix #170

**Comment by gab07** [comment]
There's a temporary [fix](https://github.com/natintosh/intl_phone_number_input/issues/155#issuecomment-753009509)

TL;DR: Just make final field holding the initial value:
`final _initialPhone = PhoneNumber(isoCode: "PA");`

and use it like this:
`InternationalPhoneNumberInput(initialValue: _initialPhone)` 

DO NOT:
` InternationalPhoneNumberInput(initialValue: PhoneNumber(isoCode: "PA")) `

The initial value should not be recreated between builds.

**Comment by jizero** [comment]
There's a temporar 22

setting only for the first time!  

(by provider)

    if(provider.initData == null){
      Locale myLocale = Localizations.localeOf(context);
      String initialCountry = myLocale.countryCode;
      PhoneNumber number = PhoneNumber(isoCode: initialCountry);
      provider.setInitData(number);
    }else{
      provider.initData = null;
    }


=================
InternationalPhoneNumberInput (
        ...
      initialValue: provider.initData,
     ..
);

**Comment by akhandpratap14** [comment]
I still cant do it. Please Help

**Comment by pbdev-MohiniThakkar** [comment]
I'm able to resolve that, please find the code snippet : 

`   Obx(() {
                                return InternationalPhoneNumberInput(
                                    textFieldController: _numberController,
                                    textStyle:
                                        AppFont.Body1_Regular(size: 12.sp),
                                    selectorTextStyle:
                                        AppFont.Body1_Regular(size: 12.sp),
                                    onInputChanged: (PhoneNumber phoneNumber) {
                                      print(phoneNumber.dialCode);
                                      print(phoneNumber.isoCode);

                                      dialCode.value = phoneNumber.dialCode!;
                                      number.value = phoneNumber.phoneNumber!;
                                      _initialNumber.value = phoneNumber;
                                    },
                                    initialValue: _initialNumber.value,
                                    selectorConfig: const SelectorConfig(
                                      selectorType:
                                          PhoneInputSelectorType.DIALOG,
                                      setSelectorButtonAsPrefixIcon: true,
                                      leadingPadding: 10,
                                    ),
                                    inputBorder: const OutlineInputBorder(),
                                    inputDecoration: InputDecoration(
                                      fillColor: ThemeColor.bgColor,
                                      filled: true,
                                      enabledBorder: OutlineInputBorder(
                                        borderRadius:
                                            BorderRadius.circular(28.0.r),
                                        borderSide: BorderSide(
                                            width: 1.w,
                                            color: ThemeColor.borderColor),
                                      ),
                                      border: OutlineInputBorder(
                                        borderRadius:
                                            BorderRadius.circular(28.0.r),
                                        borderSide: BorderSide(
                                          width: 1.w,
                                          color: ThemeColor.violetThemeColor,
                                        ),
                                      ),
                                      hintStyle: AppFont.Body1_Regular(
                                          color: ThemeColor.lightGrey,
                                          size: 12.sp),
                                      labelStyle: AppFont.Body1_Regular(
                                          color: ThemeColor.lightGrey,
                                          size: 12.sp),
                                      suffixIcon: number.value != "" &&
                                              number.value.length > 10
                                          ? Container(
                                              padding: ltrb(
                                                  bottom: 0,
                                                  left: 15.0,
                                                  right: 15.0,
                                                  top: 0),
                                              child: SvgPicture.asset(
                                                AppImages.icDone,
                                                height: 5.h,
                                                width: 5.h,
                                              ),
                                            )
                                          : null,
                                    ));
                              }),`

and Declare _initialNumber as mentioned below:

` final _initialNumber = PhoneNumber(isoCode: "KE").obs;`

Happy Coding 👍

**Comment by jamt0** [comment]
I think te problem is with the `onInputChanged `, I remove this, and the problem fix, i dont kwon why

**Comment by jamt0** [comment]
> There's a temporary [fix](https://github.com/natintosh/intl_phone_number_input/issues/155#issuecomment-753009509)
> 
> TL;DR: Just make final field holding the initial value: `final _initialPhone = PhoneNumber(isoCode: "PA");`
> 
> and use it like this: `InternationalPhoneNumberInput(initialValue: _initialPhone)`
> 
> DO NOT: `InternationalPhoneNumberInput(initialValue: PhoneNumber(isoCode: "PA"))`
> 
> The initial value should not be recreated between builds.

This works for me

**Comment by JesseLeDavis** [comment]
Is there any progress on this, Ive I've tried every solution and nothing works. If i call nothing in the onInputMethodChanged method it works but then whats the point using it.
